lib/store.rb

```diff
@@ -0,0 +1,41 @@
+require 'byebug'
+class Store
+
+  attr_reader :client
+
+  def initialize 
+    creds = Aws::Credentials.new(AwsLoader::AWS["access_key_id"], AwsLoader::AWS["secret_access_key"])
+    @client = Aws::S3::Client.new(region: AwsLoader::AWS["region"], credentials: creds)
+  end
+    
+  @@root ||= Dir.pwd
+  
+  def self.root
+    @@root
+  end
+  
+  def create_bucket(bucket_name)
+    raise S3StoreArgumentError, "S3StoreArgumentError :: Invalid bucket name.".colorize(:red) if bucket_name.nil?
+    begin
+      response = client.create_bucket({bucket: bucket_name})
+      puts "S3 Bucket #{bucket_name} created successfully!".colorize(:green) unless response.nil?
+    rescue Aws::S3::Errors::InvalidBucketName => e
```

Comment:
How about having a validation for the bucket name **before** creating it?

Reply:
I have added a name validation code in the app.rb file but commented it out for the time being. Yet to implement the regex. Validation for bucket name is a bit complicated so taking time to build the regex. Once that is done successfully, will not take time to integrate that into the application.